glusterd2/bricksplanner/planner.go

```diff
@@ -95,6 +87,37 @@ func getBricksLayout(req *api.VolCreateReq) ([]api.SubvolReq, error) {
 		return nil, err
 	}
 
+	if req.MaxBrickSize > 0 && req.MaxBrickSize < minBrickSize {
+		return nil, errors.New("invalid max-brick-size, Minimum size required is " + strconv.Itoa(minBrickSize))
+	}
+
+	// If max Brick size is specified then decide distribute
+	// count and Volume Size based on Volume Type
+	if req.MaxBrickSize > 0 && req.Size > req.MaxBrickSize {
```

Comment:
what happens if `req.MaxBrickSize` if greater than `req.Size`?

Reply:
It will be same as not passing that option

Reply:
as this is an input request we need to return an error to the user.

Reply:
I don't think that is an error. It is satisfying the user's condition of max-brick-size.